src/Navigation/INavigationEngine.ts

```diff
@@ -0,0 +1,135 @@
+import { TransformNode } from "../Meshes/transformNode";
+import { Vector3 } from "../Maths/math";
+import { AbstractMesh } from "../Meshes/abstractMesh";
+import { Mesh } from "../Meshes/mesh";
+import { Scene } from "../scene";
+
+/**
+ * Navigation plugin interface to add navigation constrained by a navigation mesh
+ */
+export interface INavigationEnginePlugin {
+    name: string;
+
+    /// Create a navigation mesh that will constrain a Crowd of agents
+    createMavMesh(mesh: AbstractMesh, parameters: NavMeshParameters): void;
+
+    /// Create and a to the scene a debug navmesh. Set the material to make it visible
+    createDebugNavMesh(scene: Scene): Mesh;
+
+    /// Get a navigation mesh constrained position, closest to the parameter position
+    getClosestPoint(position: Vector3): Vector3;
+
+    /// return a random position within a position centered disk
+    getRandomPointAround(position: Vector3, maxRadius: number): Vector3;
+
+    ///
+    isSupported(): boolean;
+
+    /// Create a new Crowd so you can add agents
+    createCrowd(maxAgents: number, maxAgentRadius: number, scene: Scene): ICrowd;
+
+    ///
+    dispose(): void;
+}
+
+/**
+ * Crowd Interface. A Crowd is a collection of moving agents constrained by a navigation mesh
+ */
+export interface ICrowd {
+    /// add a new agent to the crowd with the specified parameter a corresponding transformNode.
+    /// You can attach anything to that node. The node position is updated in the scene update tick.
+    addAgent(pos: Vector3, parameters: AgentParameters, transform: TransformNode): number;
+
+    ///
+    getAgentPosition(index: number): Vector3;
+
+    ///
+    getAgentVelocity(index: number): Vector3;
+
+    /// remove a particular agent previously created
+    removeAgent(index: number): void;
+
+    /// get the list of all agents attached to this crowd
+    getAgents() : number[];
+
+    /// Tick update done by the Scene. Agent position/velocity/acceleration is updated by this function
+    update(deltaTime: number): void;
+
+    /// Asks a particular agent to go to a destination. That destination is constrained by the navigation mesh
+    agentGoto(index: number, destination: Vector3): void;
+
+    /// dispose
+    dispose() : void;
+}
+
+export class AgentParameters {

```

Comment:
if this is supposed to be a class, probably should initialize values and put access modifier (public)

Reply:
Done with Interface